docs/content/integrations/duckdb/reference.mdx

```diff
@@ -0,0 +1,584 @@
+- [Using the DuckDB I/O manager with other I/O managers](#using-the-duckdb-io-manager-with-other-io-managers)
+- [Storing and loading PySpark and Polars DataFrames in DuckDB](#storing-and-loading-pyspark-and-polars-dataframes-in-duckdb)
+- [Storing multiple DataFrame types in DuckDB](#TODO)
+- [Executing custom SQL commands with the DuckDB resource](#executing-custom-sql-commands-with-the-duckdb-resource)
+
+---
+
+## Selecting specific columns in a downstream asset
+
+Sometimes you may not want to fetch an entire table as the input to a downstream asset. With the DuckDB I/O manager, you can select specific columns to load by supplying metadata on the downstream asset.
+
+```python file=/integrations/duckdb/downstream_columns.py
+import pandas as pd
+
+from dagster import AssetIn, asset
+
+
+# this example uses the iris_dataset asset from Step 2 of the Using Dagster with DuckDB tutorial
+
+
+@asset(
+    ins={
+        "iris_sepal": AssetIn(
+            key="iris_dataset",
+            metadata={"columns": ["sepal_length_cm", "sepal_width_cm"]},
+        )
+    }
+)
+def sepal_data(iris_sepal: pd.DataFrame) -> pd.DataFrame:
+    iris_sepal["sepal_area_cm2"] = (
+        iris_sepal["sepal_length_cm"] * iris_sepal["sepal_width_cm"]
+    )
+    return iris_sepal
+```
+
+In this example, we only use the columns containing sepal data from the `IRIS_DATASET` table created in [Step 2: Create tables in DuckDB](/integrations/duckdb/using-duckdb-with-dagster#store-a-dagster-asset-as-a-table-in-duckdb) of the [Using Dagster with DuckDB tutorial](/integrations/duckdb/using-duckdb-with-dagster). To select specific columns, we can add metadata to the input asset. We do this in the `metadata` parameter of the `AssetIn` that loads the `iris_dataset` asset in the `ins` parameter. We supply the key `columns` with a list of names of the columns we want to fetch.
+
+When Dagster materializes `sepal_data` and loads the `iris_dataset` asset using the DuckDB I/O manager, it will only fetch the `sepal_length_cm` and `sepal_width_cm` columns of the `IRIS.IRIS_DATASET` table and pass them to `sepal_data` as a Pandas DataFrame.
+
+---
+
+## Storing partitioned assets
+
+The DuckDB I/O manager supports storing and loading partitioned data. In order to correctly store and load data from the DuckDB table, the DuckDB I/O manager needs to know which column contains the data defining the partition bounds. The DuckDB I/O manager uses this information to construct the correct queries to select or replace the data. In the following sections, we describe how the I/O manager constructs these queries for different types of partitions.
+
+<TabGroup>
+<TabItem name="Static partitioned assets">
+
+### Storing static partitioned assets
+
+In order to store static partitioned assets in DuckDB, you must specify `partition_expr` metadata on the asset to tell the DuckDB I/O manager which column contains the partition data:
```

Comment:
```suggestion
To store static partitioned assets in DuckDB, specify `partition_expr` metadata on the asset to tell the DuckDB I/O manager which column contains the partition data:
```

Suggesting we remove `In order` and `you must` to make things less wordy. `You must` isn't actually needed here since this is a command and uses 'understood you'.